docs/gallery/advanced_io/plot_editing.py

```diff
@@ -0,0 +1,178 @@
+
+##############################################
+# This also works for attributes of groups:
+
+with h5py.File("test_edit.nwb", "r+") as f:
+    f["acquisition"]["synthetic_timeseries"].attrs["description"] = "Random values in volts"
+
+##############################################
+# .. warning::
+#    You can edit values that will bring the file out of compliance with
+#    the NWB specification.
+#
+# Renaming groups and datasets
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Rename groups and datasets in-place using the ``move`` method. For example, to rename
+# the ``"synthetic_timeseries"`` group:
+
+with h5py.File("test_edit.nwb", "r+") as f:
+    f["acquisition"].move("synthetic_timeseries", "synthetic_timeseries_renamed")
+
+##############################################
+# You can use this same technique to move a group or dataset to a different location in
+# the file. For example, to move the ``"synthetic_timeseries_renamed"`` group to the
+# ``"analysis"`` group:
+
+with h5py.File("test_edit.nwb", "r+") as f:
+    f["acquisition"].move("synthetic_timeseries_renamed", "/analysis/synthetic_timeseries_renamed")
+
+##############################################
+# Changing the shape of dataset
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Whether it is possible to change the shape of a dataset depends on how the dataset was
+# created. If the dataset was created with a flexible shape, then it is possible to
+# change in-place. Creating a dataset with a flexible shape is done by specifying the
+# ``maxshape`` argument of the :py:class:`~hdmf.backends.hdf5.h5_utils.H5DataIO` class
+# constructor. Using a ``None`` value for ``maxshape`` allows the dataset to be reset
+# arbitrarily long in that dimension. Chunking is required for datasets with flexible
+# shapes. Setting ``maxshape`` automatically sets chunking to ``True``, if not specified.
+#
+# First, let's create an NWB file with a dataset with a flexible shape:
+
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+nwbfile = NWBFile(
+    session_description="my first synthetic recording",
+    identifier="EXAMPLE_ID",
+    session_start_time=datetime.now(tzlocal()),
+    session_id="LONELYMTN",
+)
+
+data_io = H5DataIO(data=np.random.randn(100, 100), maxshape=(None, 100))
+
+nwbfile.add_acquisition(
+    TimeSeries(
+        name="synthetic_timeseries",
+        description="Random values",
+        data=data_io,
+        unit="m",
+        rate=10e3,
+    )
+)
+
+with NWBHDF5IO("test_edit2.nwb", "w") as io:
+    io.write(nwbfile)
+
+##############################################
+# The ``None`` in ``maxshape`` means that the dataset has an unlimited shape. You can
+# also use an integer to specify a fixed ``maxshape``. If you do not specify a
+# ``maxshape``, then the dataset will have a fixed shape. You can change the shape of
+# this dataset.
+
+import h5py
+
+with h5py.File("test_edit2.nwb", "r+") as f:
+    f["acquisition"]["synthetic_timeseries"]["data"].resize((200, 100))
+
```

Comment:
Same comment as above, I think we could go through PyNWB here, since it exposes the h5py datasets, i.e,. I think the following should work

```
with NWBHDF5IO("test_edit2.nwb", "a") as io:
    nwbfile = io.read()
    nwbfile.get_acquisition("synthetic_timeseries").resize((200, 100))
```

Reply:
yes that should work